projects/mercury/src/metadata/common/LinkedDataAPI.js

```diff
@@ -53,6 +53,26 @@ class LinkedDataAPI {
             .then(normalizeTypes)
             .catch(() => null);
     }
+
+    parseHierachyNodes(json) {
+        if (!json) {
+            throw Error(`Cannot parse empty response`);
+        }
+
+        return json.map(hierarchyItem => {
+            const nodeType = hierarchyItem.TypeName;
+            const children = hierarchyItem.ChildNodes;
+            const isRoot = hierarchyItem.IsRoot;
+            return {nodeType, children, isRoot};
+        });
+    }
+
+    getHierarchyNodes() {
```

Comment:
this should be moved to VocabularyAPI.js. The code in LinkedDataAPI is shared between vocabularyAPI and metadataAPI (both APIs should have an endpoint like this, this one is vocabulary specific).